core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -611,7 +611,7 @@ class Ruleset {
             for (building in era.settlerBuildings)
                 if (building !in buildings)
                     lines += "Nonexistent building $building built by settlers when starting in ${era.name}"
-            if (era.startingMilitaryUnitCount != 0 && era.startingMilitaryUnit !in units)
+            if (era.startingMilitaryUnitCount != 0 && era.startingMilitaryUnit != Constants.eraSpecificUnit && era.startingMilitaryUnit !in units)
```

Comment:
I think this check is redundant? If we need to add `Constants.eraSpecificUnit`, it is actualy replaced with era.startingMilitaryUnit?

Reply:
thanks for the warning

Reply:
Then I don't understand your previous comment.

Reply:
What I meant is that only checking whether the era specific unit exists if that unit is used in the era is incorrect, as it is possible that it is not used in the era's, but is used in by AI's on higher difficulties, meaning that also in those cases the unit should still exist